backend/spotihue/tasks.py

```diff
@@ -11,21 +11,65 @@
 logger = logging.getLogger(__name__)
 
 
+def is_spotihue_running() -> bool:
+    """Queries for the SpotiHue task by its cached spotihue_task_id. There should only be one
+    run_spotihue task running at any given time.
+
+    Returns:
+         bool: Whether the SpotiHue (run_spotihue) task is running.
+    """
+    celery_inspect = celery_app.control.inspect()
+    spotihue_task_id = redis_client.get(constants.REDIS_SPOTIHUE_TASK_ID)
+
+    if spotihue_task_id:
+        spotihue_task_id = spotihue_task_id.decode('utf-8')
+
+        task_query_result = celery_inspect.query_task(*[spotihue_task_id, ])
+        celery_host = list(task_query_result.keys())[0]  # we are only using the single celery worker
+
+        if task_query_result[celery_host].get(spotihue_task_id) is not None:
+            return True
+
+    return False
+
+
+@celery_app.task
+def clear_spotihue_task_id(*args, **kwargs):
+    logger.info('Clearing spotihue_task_id')
+    redis_client.delete(constants.REDIS_SPOTIHUE_TASK_ID)
+
+
 @celery_app.task
-def run_spotihue(lights: List[str]) -> None:
+def run_spotihue(lights: List[str], current_track_retries: int = 0) -> None:
     logger.info("Running spotihue")
     spotihue.change_all_lights_to_normal_color(lights)
 
-    while spotihue.ascertain_track_playing():
+    retries_spent = 0
+    retry_wait_seconds = 5
+
+    while True:
+        track_info = spotihue.retrieve_current_track_information(supply_defaults=False)
+
+        # if user's Spotify has no current track (playing or paused), we wait to get a current track
```

Comment:
Can we make the default a larger number? I was envisioning that the spotihue task would keep checking until the user physically stopped spotihue. However, I think adding in retries is smart in case something happens and we don't want it running forever. On the flip side, I can see a user pausing a song for a while and then expecting spotihue to keep working when the start playing a song again.

Reply:
What do you think about making this task a periodic task? That way, we could have it execute every 5 seconds (or whatever short time interval you prefer), and work in the exact same way that it's already working, only minus the `while True` loop. it'd be a little more celery-esque, especially (I believe) when it comes to terminating the task (https://stackoverflow.com/questions/52724789/which-is-the-best-way-to-programatically-terminate-cancel-a-celery-task <-- our task right now looks a bit more like it should maybe extend [AbortableTask](https://docs.celeryq.dev/en/latest/reference/celery.contrib.abortable.html))

Reply:
But also, I upped our current_track_retries from 3 to 10 in `start_spotihue()` 👍

Reply:
Let's capture this in an issue.